internal/providers/terraform/aws/rds_cluster.go

```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 	backupStorageRetention := decimal.NewFromInt(d.Get("backup_retention_period").Int())
 	if backupStorageRetention.GreaterThan(decimal.NewFromInt(1)) {
 
-		snapShotStorageSizeGB := decimal.Zero
+		var snapShotStorageSizeGB *decimal.Decimal
 		if u != nil && u.Get("backup_snapshot_size_gb").Exists() {
-			snapShotStorageSizeGB = decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float())
-		}
+			snapShotStorageSizeGB = decimalPtr(decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float()))
+
+			totalBackupStorageGB := calculateBackupStorage(*snapShotStorageSizeGB, backupStorageRetention)
+			if totalBackupStorageGB.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBackupStorageCostComponent(region, &totalBackupStorageGB, databaseEngine))
+			}
+		} else {
+			var unknown *decimal.Decimal
 
-		totalBackupStorageGB := calculateBackupStorage(snapShotStorageSizeGB, backupStorageRetention)
-		if totalBackupStorageGB.GreaterThan(decimal.Zero) {
-			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, totalBackupStorageGB, databaseEngine))
+			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, unknown, databaseEngine))
 		}
 	}
 
 	if databaseEngineMode != "Serverless" && !strings.Contains(d.Get("engine").String(), "postgresql") {
-		averageStatements := decimal.Zero
-		if u != nil && u.Get("average_statements_per_hr").Exists() {
-			averageStatements = decimal.NewFromInt(u.Get("average_statements_per_hr").Int())
-		}
+		var averageStatements, backtrackChangeRecords, backtrackWindowHours *decimal.Decimal
+		if u != nil && averageStatementsPerHrExists(u) && changeRecordsPerStatementExists(u) && backtrackWindowHrsExists(u) {
+			averageStatements = decimalPtr(decimal.NewFromInt(u.Get("average_statements_per_hr").Int()))
+			backtrackChangeRecords = decimalPtr(decimal.NewFromFloat(u.Get("change_records_per_statement").Float()))
+			backtrackWindowHours = decimalPtr(decimal.NewFromInt(u.Get("backtrack_window_hrs").Int()))
 
-		backtrackChangeRecords := decimal.Zero
-		if u != nil && u.Get("change_records_per_statement").Exists() {
-			backtrackChangeRecords = decimal.NewFromFloat(u.Get("change_records_per_statement").Float())
-		}
+			totalBacktrackChangeRecords := calculateBacktrack(*averageStatements, *backtrackChangeRecords, *backtrackWindowHours)
 
-		backtrackWindowHours := decimal.Zero
-		if u != nil && u.Get("backtrack_window_hrs").Exists() {
-			backtrackWindowHours = decimal.NewFromInt(u.Get("backtrack_window_hrs").Int())
-		}
+			if totalBacktrackChangeRecords.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBacktrackCostComponent(region, &totalBacktrackChangeRecords))
+			}
+		} else {
+			var unknown *decimal.Decimal
```

Comment:
The same thing as above, `totalBacktrackChangeRecords` in `var` and use it after `if` block.
```suggestion
		} 
		costComponents = append(costComponents, auroraBacktrackCostComponent(region, totalBacktrackChangeRecords))
```